### PR TITLE
feat(welcome): add SQL snippets to saved queries card

### PR DESCRIPTION
**Pull request by pkdotson**
### SUMMARY
This Pr update the savedqueries card to the new proposed layout on welcome page. Also, updated media queries to keep text from expanding the cards to weird widths.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
Before:
Screen Shot 2020-11-12 at 1 55 56 PM

After: 
Screen Shot 2020-11-16 at 12 00 18 PM

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [x ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by nytai** [comment]
FYI, I discovered https://github.com/apache/incubator-superset/pull/11679 while testing this. You'll want to handle the null case in this PR too

**Comment by nytai** [comment]
Screen Shot 2020-11-12 at 12 36 28 PM

Also, it looks card size changes based on the content. I'm not sure whether this is desirable or not. I just worked on query history page added a truncation function you might be interested in 

https://github.com/apache/incubator-superset/blob/432e5ab4607fe4743562101336c63dcfd18b0867/superset-frontend/src/views/CRUD/data/query/QueryList.tsx#L52

**Comment by junlincc** [comment]
adding tooltip for long title display on hover. don't merge yet @rusackas

**Comment by junlincc** [comment]
Screen Shot 2020-11-17 at 10 04 09 PM

now im able to view open PR finally......
1. can we get rid of the default Note in the query card 
2. what is "3"? it's not part of the query
3. add tooltip to show entire title

**Comment by junlincc** [comment]
Screen Shot 2020-11-23 at 7 55 44 PM
Screen Shot 2020-11-23 at 7 56 32 PM


LGTM, @rusackas let's merge!

**Review by junlincc — APPROVED** [review]

File: superset-frontend/src/views/CRUD/utils.tsx [hunk 1]
```diff
@@ -261,22 +261,31 @@ export function handleDashboardDelete(
   );
 }
 
+export function shortenSQL(sql: string, maxLines: number) {
+  let lines: string[] = sql.split('\n');
+  if (lines.length >= maxLines) {
+    lines = lines.slice(0, maxLines);
+    lines.push('...');
```
Comment (ktmud) [inline superset-frontend/src/views/CRUD/utils.tsx:268]: The `...` is probably not necessary, since it will be pushed down to hidden anyway if any of the SQL lines was wrapped:

image

Can probably just set a higher `maxLines` and make sure there's no unnecessary blank space at the bottom of the SQL code area.

Comment (rusackas) [reply superset-frontend/src/views/CRUD/utils.tsx:268]: if the line-wrapping is causing the extra lines, I would be open to any of these options:
1) Not worrying about it (though a super-duper long line might get crazy)
2) Not letting it wrap (CSS e.g. `white-space: nowrap`)
3) Cropping off the query with a max-height
4) ... dare I say scrolling?

Personally I'd vote for #2 or #2

File: superset-frontend/images/empty-query.svg [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+specific language governing permissions and limitations
+under the License.
+-->
+<svg width="458" height="146" viewBox="0 0 458 146" fill="none" xmlns="http://www.w3.org/2000/svg">
+<path d="M1 5C1 2.79086 2.79086 1 5 1H453C455.209 1 457 2.79086 457 5V145H1V5Z" fill="#D9DBE4" stroke="#D9DBE4"/>
+<path d="M17 21C17 18.7909 18.7909 17 21 17H125C127.209 17 129 18.7909 129 21V29C129 31.2091 127.209 33 125 33H21C18.7909 33 17 31.2091 17 29V21Z" fill="#ECEEF2"/>
+<path d="M17 45C17 42.7909 18.7909 41 21 41H223C225.209 41 227 42.7909 227 45V53C227 55.2091 225.209 57 223 57H21C18.7909 57 17 55.2091 17 53V45Z" fill="#ECEEF2"/>
+<path d="M21 65C18.7909 65 17 66.7909 17 69V77C17 79.2091 18.7909 81 21 81H172C174.209 81 176 79.2091 176 77V69C176 66.7909 174.209 65 172 65H21Z" fill="#ECEEF2"/>
+<path d="M17 93C17 90.7909 18.7909 89 21 89H286C288.209 89 290 90.7909 290 93V101C290 103.209 288.209 105 286 105H21C18.7909 105 17 103.209 17 101V93Z" fill="#ECEEF2"/>
+<path d="M21 113C18.7909 113 17 114.791 17 117V125C17 127.209 18.7909 129 21 129H73C75.2091 129 77 127.209 77 125V117C77 114.791 75.2091 113 73 113H21Z" fill="#ECEEF2"/>
```
Comment (rusackas) [inline superset-frontend/images/empty-query.svg:25]: We should set all of these color values to `fill="currentColor"`

File: superset-frontend/src/views/CRUD/welcome/SavedQueries.tsx [hunk 3]
```diff
@@ -57,19 +64,45 @@ interface SavedQueriesProps {
...
   }
-  .holder {
-    margin: ${({ theme }) => theme.gridUnit * 2}px;
+  .query-title {
+    padding: ${({ theme }) => theme.gridUnit * 2 + 2}px;
+    font-size: ${({ theme }) => theme.typography.sizes.l}px;
   }
 `;
+
+const ShowEmptyState = styled.div`
```
Comment (rusackas) [inline superset-frontend/src/views/CRUD/welcome/SavedQueries.tsx:87]: Is there a reason not to use the <Empty> component like we're doing in other sections?

File: superset-frontend/src/views/CRUD/welcome/SavedQueries.tsx [hunk 4]
```diff
@@ -57,19 +64,45 @@ interface SavedQueriesProps {
...
+  width: 100%;
+  height: 171px;
+  background-repeat: no-repeat;
+  background-size: 539px;
+`;
+
+const QueryContainer = styled.div`
+  pre {
+    height: ${({ theme }) => theme.gridUnit * 40}px;
+    border: none !important;
```
Comment (rusackas) [inline superset-frontend/src/views/CRUD/welcome/SavedQueries.tsx:98]: Ping me when you have a chance, we can pair up and see if there's a better selector to remove the need for `!important` here.

File: superset-frontend/src/views/CRUD/utils.tsx (hunk 1 above)
Comment (ktmud) [reply superset-frontend/src/views/CRUD/utils.tsx:268]: Personally I'd say let it wrap and just get rid of "...".

File: superset-frontend/src/views/CRUD/welcome/SavedQueries.tsx (hunk 3 above)
Comment (pkdotson) [reply superset-frontend/src/views/CRUD/welcome/SavedQueries.tsx:87]: The way the logic is set for the others  it checks  if there is thumbnail url where here we are checking if if there is a query. The way that I'm using the card is a little different from the chart and dashboard since we are depeneding on whether there is a block code of sql for the reactsyntaxhighlighe to show.

File: superset-frontend/images/empty-query.svg (hunk 2 above)
Comment (nytai) [reply superset-frontend/images/empty-query.svg:25]: Do we want that? This is being used as an image and not an icon.

File: superset-frontend/src/views/CRUD/welcome/SavedQueries.tsx [hunk 5]
```diff
@@ -57,19 +64,51 @@ interface SavedQueriesProps {
...
+    border-bottom: 1px solid ${({ theme }) => theme.colors.grayscale.light2};
+    & > div {
+      height: 171px;
+    }
+  }
+  .gradient-container > div {
+    background-size: contain;
+    background-repeat: no-repeat;
+    background-position: center;
+    background-color: #d8dbe4;
```
Comment (rusackas) [inline superset-frontend/src/views/CRUD/welcome/SavedQueries.tsx:82]:
```suggestion
    background-color: ${({ theme }) => theme.colors.grayscale.light3};
```

File: superset-frontend/src/views/CRUD/welcome/SavedQueries.tsx [hunk 6]
```diff
@@ -57,19 +64,51 @@ interface SavedQueriesProps {
...
 
+export const CardStyles = styled.div`
+  cursor: pointer;
+  a {
+    text-decoration: none;
+  }
+  .ant-card-cover {
+    border-bottom: 1px solid ${({ theme }) => theme.colors.grayscale.light2};
+    & > div {
+      height: 171px;
```
Comment (rusackas) [inline superset-frontend/src/views/CRUD/welcome/SavedQueries.tsx:75]: This can be an experiment for another PR, but if this card is basically a text box, we may be able to base its height on the number of lines set elsewhere in the JS. That is, if `line-height` is pulled from the Theme variables, the height could be `numLines * lineHeight`. Anyway, something to fiddle with to make this a little more dynamic.

**Review by rusackas — CHANGES_REQUESTED** [review]
Lookin' good to me, just one little nit to fix (I made a code suggestion)

File: superset-frontend/images/empty-query.svg [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+<svg width="458" height="146" viewBox="0 0 458 146" fill="none" xmlns="http://www.w3.org/2000/svg">
+<path d="M1 5C1 2.79086 2.79086 1 5 1H453C455.209 1 457 2.79086 457 5V145H1V5Z" fill="#D9DBE4" stroke="#D9DBE4"/>
```
Comment (rusackas) [inline superset-frontend/images/empty-query.svg:20]: I think we can get rid of this <path> so that it's effectively transparent.

**Review by rusackas — APPROVED** [review]

**Comment by junlincc** [comment]
#11659
